### PR TITLE
support for multiple volume groups

### PR DESCRIPTION
**Pull request by zoetrope**
refs: #125 #131

### CODE REVIEW

**Comment by deveshk0** [comment]
@zoetrope  still working on that? Any idea by when we can get this?

**Comment by zoetrope** [comment]
@deveshk0 
I'm currently working on.
I think I can release it within a couple of weeks.

File: docs/topolvm-scheduler.md [hunk 1]
```diff
@@ -19,15 +19,15 @@ Scheduler policy
...
+          "name": "topolvm.io/capacity",
           "ignoredByScheduler": true
         }],
         "nodeCacheCapable": false
     }]
 }
 ```
 
-As it shown, only pods that request `topolvm.cybozu.com/capacity` resource are
+As it showed, only pods that request `topolvm.io/capacity` resource are
```
Comment (ymmt2005) [inline docs/topolvm-scheduler.md:30]:
```suggestion
As shown, only pods that request `topolvm.io/capacity` resource are
```

File: docs/topolvm-scheduler.md [hunk 2]
```diff
@@ -53,10 +53,24 @@ This verb scores nodes.  The score of a node is calculated by this formula:
...
 ------------------
 
-| Name      | Type    | Default | Description                 |
-| --------- | ------- | ------: | --------------------------- |
-| `listen`  | string  | `:8000` | HTTP listening address      |
-| `divisor` | float64 |       1 | A variable for node scoring |
+| Name      | Type    | Default                       | Description            |
+| --------- | ------- | ----------------------------: | ---------------------- |
+| `listen`  | string  | `:8000`                       | HTTP listening address |
+| `config`  | float64 | `/etc/topolvm/scheduler.yaml` | Config file path       |
```
Comment (ymmt2005) [inline docs/topolvm-scheduler.md:64]:
```suggestion
| `config`  | string | `/etc/topolvm/scheduler.yaml` | Config file path       |
```

File: driver/controller.go [hunk 3]
```diff
@@ -351,8 +364,13 @@ func (s controllerService) ControllerExpandVolume(ctx context.Context, req *csi.
 			NodeExpansionRequired: true,
 		}, nil
 	}
-
-	capacity, err := s.nodeService.GetCapacityByName(ctx, lv.Spec.NodeName)
+	var vgName string
+	if len(lv.Spec.VGName) == 0 {
```
Comment (ymmt2005) [inline driver/controller.go:368]: As mentioned above, this should not be empty.

File: driver/node.go [hunk 4]
```diff
@@ -34,19 +35,23 @@ const (
 var nodeLogger = logf.Log.WithName("driver").WithName("node")
 
 // NewNodeService returns a new NodeServer.
-func NewNodeService(nodeName string, conn *grpc.ClientConn) csi.NodeServer {
+func NewNodeService(nodeName, defaultVG string, conn *grpc.ClientConn, service *k8s.LogicalVolumeService) csi.NodeServer {
```
Comment (ymmt2005) [inline driver/node.go:38]: NodeService should not care about default VG.
Only the controller should care.

**Comment by deveshk0** [comment]
Any Update on this??

**Comment by ymmt2005** [comment]
The work is going under another branch.
https://github.com/cybozu-go/topolvm/commits/device-class

@zoetrope Do you merge changes in `device-class` into this?

**Comment by zoetrope** [comment]
I recreated this PR https://github.com/cybozu-go/topolvm/pull/147

**Pull request closed without merging**